--- a/PKGBUILD	2021-01-20 08:12:45.133051278 +0100
+++ b/PKGBUILD	2021-01-20 08:15:38.514634925 +0100
@@ -1,5 +1,6 @@
 # Maintainer: graysky <graysky AT archlinux DOT us>
 # Contributor: Jan Alexander Steffens (heftig) <jan.steffens@gmail.com>
+# Modified by Salonia Matteo <saloniamatteo@protonmail.com>
 
 ### BUILD OPTIONS
 # Set these variables to ANYTHING that is not null to enable them
@@ -49,7 +50,8 @@
 #  32. Generic-x86-64 (GENERIC_CPU)
 #  33. Native optimizations autodetected by GCC (MNATIVE)
 
-_subarch=
+# Haswell
+_subarch=23
 
 # Compile ONLY used modules to VASTLYreduce the number of modules built
 # and the build time.
@@ -59,7 +61,7 @@
 # This PKGBUILD read the database kept if it exists
 #
 # More at this wiki page ---> https://wiki.archlinux.org/index.php/Modprobed-db
-_localmodcfg=
+#_localmodcfg=
 
 ### IMPORTANT: Do no edit below this line unless you know what you're doing
 
@@ -84,19 +86,35 @@
   0001-ZEN-Add-sysctl-and-CONFIG-to-disallow-unprivileged-C.patch
   0002-HID-quirks-Add-Apple-Magic-Trackpad-2-to-hid_have_sp.patch
   0003-iwlwifi-Fix-regression-from-UDP-segmentation-support.patch
+  "modprobed.db"
+  "tune-config"
 )
 validpgpkeys=(
   'ABAF11C65A2970B130ABE3C479BE3E4300411886'  # Linus Torvalds
   '647F28654894E3BD457199BE38DBBDC86092693E'  # Greg Kroah-Hartman
 )
-b2sums=('faedb4032fd709d3f0089d706232ec0dcfdf3817223aa910112e6cd58bffea20a3127fee407a465fa3b4db1a54050fabd839809c404492820216fadae70885b9'
+b2sums=(
+        # Kernel source (linux-X.XX.X.tar.xz)
+        'faedb4032fd709d3f0089d706232ec0dcfdf3817223aa910112e6cd58bffea20a3127fee407a465fa3b4db1a54050fabd839809c404492820216fadae70885b9'
+        # Kernel source signature (linux-X.XX.X.tar.sign)
         'SKIP'
-        '37df0c9fc2214da9e94f0fb38ddc9c1f5bcf6279cacf918166efc3a0a8a9433ee10a95232992a108944160cf020fc3c240ef2fb8023581dc9cac11f973d7985d'
+        # Kernel configuration (config)
+        '78f36713cbac4f4abfea6474e7ae3bf7fe95b1c3cb43c3990f966c138a567021686d6364505a277423c9f19d36e7ec1497010e0da4e5f915b2c80159d15a97f4'
+        # Enable additional CPU optimizations patch (enable_additional_cpu_optimizations-YYYYMMDD.tar.gz)
         '7f1eb5938472f57748216bd00e0c875feab99fc1c5cb89babfea467ee30ca5c8e9fc5a691efe2e602bef1ea79820c5383822d7cec354b48d23321ccda8ee8127'
+        # CK patchset (patch-X.XX-ck1.xz)
         '067f3389124fdd937ca69e9e9568b1b3194791960a093e81037051eb6d25e80b40bf7f60c61373ac9e92bff9db760766009b1e6f9ee8429a883bb7fce2d60f8a'
+        # 0001-ZEN-Add-sysctl-and-CONFIG-to-disallow-unprivileged-C.patch
         'b42730a806a63dbf905f448a3282dec72a950ef7d16a3531d977dcae3b5c2685a5c5dd10a58e345d57084a6212353dc2f2f0102021d13c1557092e564f81eaf5'
+        # 0002-HID-quirks-Add-Apple-Magic-Trackpad-2-to-hid_have_sp.patch
         'a5dea5bb6df6fb517009ad5b104c2ea8e93ed880393a1412ca2d7fc8047abccc9c72076e40d1d4133cadc9048040846cc9bdc91e30c4d601ad5963d13498503e'
-        '9a8723104239ba4646dd0c76c5a60128892954dd2faee69ca5c5d1251c9694de8fd528ca063a2add1401ad1375ad0dcc2560ca03a1bfd03b161d34ccd25b686a')
+        # 0003-iwlwifi-Fix-regression-from-UDP-segmentation-support.patch
+        '9a8723104239ba4646dd0c76c5a60128892954dd2faee69ca5c5d1251c9694de8fd528ca063a2add1401ad1375ad0dcc2560ca03a1bfd03b161d34ccd25b686a'
+        # module database (modprobed.db)
+        '3e52ade633ffb1db3211aaa2058173482e1546fcc9ad66cd25ff56d6a9b1106fa185b8a578f4b753b146205dfc42bddb4e2d2055326eaaf8ee7178db2d1f5984'
+        # configuration modifier (tune-config)
+        'f18a5df56b415d702ae958eabe3227a296bf4fc45c91e3e505ff543beb10e59312cf247f641c94c932127aa13b24cef76242d24c7fb4558fa0a7d236a99543d7'
+)
 
 export KBUILD_BUILD_HOST=archlinux
 export KBUILD_BUILD_USER=$pkgbase
@@ -144,6 +162,9 @@
   # https://bugzilla.kernel.org/show_bug.cgi?id=207173#c6
   scripts/config --disable CONFIG_KVM_WERROR
 
+  # apply optimizations
+  ../tune-config
+
   # fix naming schema in EXTRAVERSION of ck patch set
   sed -i -re "s/^(.EXTRAVERSION).*$/\1 = /" "../${_ckpatch}"
 
@@ -168,17 +189,18 @@
     make oldconfig
   fi
 
-  ### Optionally load needed modules for the make localmodconfig
+  ### Load needed modules for the make localmodconfig
   # See https://aur.archlinux.org/packages/modprobed-db
-    if [ -n "$_localmodcfg" ]; then
-      if [ -f $HOME/.config/modprobed.db ]; then
-        echo "Running Steven Rostedt's make localmodconfig now"
-        make LSMOD=$HOME/.config/modprobed.db localmodconfig
-      else
-        echo "No modprobed.db data found"
-        exit
-      fi
-    fi
+  echo "Running Steven Rostedt's make localmodconfig now"
+
+  if [ -f "$HOME/.config/modprobed.db" ]; then
+    echo "Using modprobed.db in $HOME/.config! "
+    make LSMOD="$HOME/.config/modprobed.db" localmodconfig
+  else if [ -f "${srcdir}"/"modprobed.db" ]; then
+    echo "Using modprobed.db in ${pkgdir}! "
+    make LSMOD="${srcdir}/modprobed.db" localmodconfig
+  fi
+  fi
 
   make -s kernelrelease > version
   echo "Prepared $pkgbase version $(<version)"
@@ -194,7 +216,7 @@
 
 build() {
   cd linux-${pkgver}
-  make all
+  make all -j$(nproc)
 }
 
 _package() {
