--- ../../aur/linux-ck-upstream/PKGBUILD	2021-01-28 08:16:28.925392621 +0100
+++ PKGBUILD	2021-01-28 08:16:09.065193718 +0100
@@ -1,5 +1,6 @@
 # Maintainer: graysky <graysky AT archlinux DOT us>
 # Contributor: Jan Alexander Steffens (heftig) <jan.steffens@gmail.com>
+# Modified by Salonia Matteo <saloniamatteo@protonmail.com>
 
 ### BUILD OPTIONS
 # Set these variables to ANYTHING that is not null to enable them
@@ -49,7 +50,8 @@
 #  32. Generic-x86-64 (GENERIC_CPU)
 #  33. Native optimizations autodetected by GCC (MNATIVE)
 
-_subarch=
+# Haswell
+_subarch=23
 
 # Compile ONLY used modules to VASTLYreduce the number of modules built
 # and the build time.
@@ -59,7 +61,7 @@
 # This PKGBUILD read the database kept if it exists
 #
 # More at this wiki page ---> https://wiki.archlinux.org/index.php/Modprobed-db
-_localmodcfg=
+#_localmodcfg=
 
 ### IMPORTANT: Do no edit below this line unless you know what you're doing
 
@@ -85,20 +87,37 @@
   0002-HID-quirks-Add-Apple-Magic-Trackpad-2-to-hid_have_sp.patch
   0003-iwlwifi-Fix-regression-from-UDP-segmentation-support.patch
   0004-HID-wacom-Correct-NULL-dereference-on-AES-pen-proxim.patch
+  "modprobed.db"
+  "tune-config"
 )
 validpgpkeys=(
   'ABAF11C65A2970B130ABE3C479BE3E4300411886'  # Linus Torvalds
   '647F28654894E3BD457199BE38DBBDC86092693E'  # Greg Kroah-Hartman
 )
-b2sums=('f2ffb77efeab44e5ac74f275f1c728618a0893c752018946a3908a34dca2797982efb6646df1350f31c9cab2b780aca871dff82b63e2ec59e3bfcdafa0457581'
+b2sums=(
+        # Kernel source (linux-X.XX.X.tar.xz)
+        'f2ffb77efeab44e5ac74f275f1c728618a0893c752018946a3908a34dca2797982efb6646df1350f31c9cab2b780aca871dff82b63e2ec59e3bfcdafa0457581'
+        # Kernel source signature (linux-X.XX.X.tar.sign)
         'SKIP'
-        '37df0c9fc2214da9e94f0fb38ddc9c1f5bcf6279cacf918166efc3a0a8a9433ee10a95232992a108944160cf020fc3c240ef2fb8023581dc9cac11f973d7985d'
+        # Kernel configuration (config)
+        '78f36713cbac4f4abfea6474e7ae3bf7fe95b1c3cb43c3990f966c138a567021686d6364505a277423c9f19d36e7ec1497010e0da4e5f915b2c80159d15a97f4'
+        # Enable additional CPU optimizations patch (enable_additional_cpu_optimizations-YYYYMMDD.tar.gz)
         '7f1eb5938472f57748216bd00e0c875feab99fc1c5cb89babfea467ee30ca5c8e9fc5a691efe2e602bef1ea79820c5383822d7cec354b48d23321ccda8ee8127'
+        # CK patchset (patch-X.XX-ck1.xz)
         '067f3389124fdd937ca69e9e9568b1b3194791960a093e81037051eb6d25e80b40bf7f60c61373ac9e92bff9db760766009b1e6f9ee8429a883bb7fce2d60f8a'
+        # 0001-ZEN-Add-sysctl-and-CONFIG-to-disallow-unprivileged-C.patch
         'de81e92b92ba697e472d6397faa425ebcc09bede6c97008dcae47efd341d2eaa12ec72209c4cc97155df01358e1fce3a86d3f0b5441ac91615037de72105defc'
+        # 0002-HID-quirks-Add-Apple-Magic-Trackpad-2-to-hid_have_sp.patch
         '9450bd3d4acfa94723eaf22f2c0a13ae3bdb1ceb15ad19d1a1de2fc8d9a7b9b8f759006fd5b4286da135c68bac75791b4bf767ece8b6c40aff95799296ee0327'
+        # 0003-iwlwifi-Fix-regression-from-UDP-segmentation-support.patch
         'ea8e11fa3e9ccffe26366c2c279e11ffd4cb7ed0500028afef083188d46a0df2487800417914e63b8ac101b6a28b6a61aad91bea0b7a0305d64be3445a4f68a0'
-        'b57086457843e75899772ec37a469f524d7f5df5a1ba27ab35a97b478ed37b8deb29371db3f69b33ec10eaf38fd554cef0da7c64c451b11db84e4dc497f8bae9')
+        # 0004-HID-wacom-Correct-NULL-dereference-on-AES-pen-proxim.patch
+        'b57086457843e75899772ec37a469f524d7f5df5a1ba27ab35a97b478ed37b8deb29371db3f69b33ec10eaf38fd554cef0da7c64c451b11db84e4dc497f8bae9'
+        # module database (modprobed.db)
+        '3e52ade633ffb1db3211aaa2058173482e1546fcc9ad66cd25ff56d6a9b1106fa185b8a578f4b753b146205dfc42bddb4e2d2055326eaaf8ee7178db2d1f5984'
+        # configuration modifier (tune-config)
+        'f18a5df56b415d702ae958eabe3227a296bf4fc45c91e3e505ff543beb10e59312cf247f641c94c932127aa13b24cef76242d24c7fb4558fa0a7d236a99543d7'
+)
 
 export KBUILD_BUILD_HOST=archlinux
 export KBUILD_BUILD_USER=$pkgbase
@@ -146,6 +165,9 @@
   # https://bugzilla.kernel.org/show_bug.cgi?id=207173#c6
   scripts/config --disable CONFIG_KVM_WERROR
 
+  # apply optimizations
+  ../tune-config
+
   # fix naming schema in EXTRAVERSION of ck patch set
   sed -i -re "s/^(.EXTRAVERSION).*$/\1 = /" "../${_ckpatch}"
 
@@ -170,17 +192,18 @@
     make oldconfig
   fi
 
-  ### Optionally load needed modules for the make localmodconfig
+  ### Load needed modules for the make localmodconfig
   # See https://aur.archlinux.org/packages/modprobed-db
-    if [ -n "$_localmodcfg" ]; then
-      if [ -f $HOME/.config/modprobed.db ]; then
-        echo "Running Steven Rostedt's make localmodconfig now"
-        make LSMOD=$HOME/.config/modprobed.db localmodconfig
-      else
-        echo "No modprobed.db data found"
-        exit
-      fi
-    fi
+  echo "Running Steven Rostedt's make localmodconfig now"
+
+  if [ -f "$HOME/.config/modprobed.db" ]; then
+    echo "Using modprobed.db in $HOME/.config! "
+    make LSMOD="$HOME/.config/modprobed.db" localmodconfig
+  else if [ -f "${srcdir}"/"modprobed.db" ]; then
+    echo "Using modprobed.db in ${pkgdir}! "
+    make LSMOD="${srcdir}/modprobed.db" localmodconfig
+  fi
+  fi
 
   make -s kernelrelease > version
   echo "Prepared $pkgbase version $(<version)"
@@ -196,7 +219,7 @@
 
 build() {
   cd linux-${pkgver}
-  make all
+  make all -j$(nproc)
 }
 
 _package() {
